### ISSUE TITLE
GX ignores the result_format parameter when running a Checkpoint

### ISSUE THREAD

**Issue by toivomattila** [great-expectations/great_expectations#8469]
**Describe the bug**
I've run across an issue where GX seems to ignore the `result_format  `parameter.
I've set the `result_format `for an expectation to `"BOOLEAN_ONLY"` but sensitive information is still shown in checkpoint results & Data Docs.

**What I've done**
I have a script (below) with `result_format` set to `"BOOLEAN_ONLY"`
After running a checkpoint against invalid data, I checked the results and found the data in the results even though it should be excluded.
The data is included in the partial_unexpected_list  section in the results.
The partial_unexcepted_list  contents are also shown in the Data Docs

**To Reproduce**
Please include your great_expectations.yml config, the code you’re executing that causes the issue, and the full stack trace of any error(s).

Code to reproduce:
```
import great_expectations as gx

import pandas as pd

context = gx.get_context()

df = pd.DataFrame({"secret": ["a"]})

datasource = context.sources.add_pandas(name="secret_datasource")
data_asset = datasource.add_dataframe_asset(name="secret_asset")
batch_request = data_asset.build_batch_request(dataframe=df)

expectation_suite = context.add_expectation_suite(
    expectation_suite_name="secret_expectations",
    expectations=[
        {
          "expectation_type": "expect_column_values_to_match_regex",
          "kwargs": {
            "column": "secret",
            "regex": "\d",
            "result_format": {
              "result_format": "BOOLEAN_ONLY"
            }
          },
          "meta": {}
        },
    ],
)

checkpoint = context.add_checkpoint(
    name="secret_checkpoint",
    validations=[
        {
            "batch_request": batch_request,
            "expectation_suite_name": "secret_expectations",
        }
    ],
)

result = checkpoint.run()

print(result.run_results[list(result.run_results.keys())[0]]["validation_result"]["results"][0]["result"]["partial_unexpected_list"])

#context.build_data_docs()
#context.open_data_docs()
```

Output from running the script:
```
['a']
```

**Expected behavior**
I expected the partial_unexpected_list to be empty or not be in the results that the checkpoint outputs
Additionally, I expected the information to not be in Data Docs
Instead, the data is that didn't fit the expectation is included in the results

**Environment (please complete the following information):**
 - Operating System: Ubuntu Linux (in WSL)
 - Great Expectations Version: 0.17.7 (also tested with versions 0.16.15-0.17.7)
 - Datasource: in-memory DataFrame, also tested with filesystem CSV
 - Cloud environment: local, didn't test any cloud environment

**Additional context**
GX didn't output any logs or throw any warnings or errors
Followed the docs here to write the expectation: https://docs.greatexpectations.io/docs/reference/expectations/result_format/

Additionally, there's a mismatch between the expectation suite and an expectation suite that a validator returns.
In the code above, printing the `expectation_suite` variable gives:
```
{
  "expectation_suite_name": "secret_expectations",
  "ge_cloud_id": null,
  "expectations": [
    {
      "expectation_type": "expect_column_values_to_match_regex",
      "kwargs": {
        "column": "secret",
        "regex": "\\d",
        "result_format": {
          "result_format": "BOOLEAN_ONLY"
        }
      },
      "meta": {}
    }
  ],
  "data_asset_type": null,
  "meta": {
    "great_expectations_version": "0.17.7"
  }
}
```

But getting the expectation suite out of a validator by running
```
validator = context.get_validator(
    expectation_suite_name="secret_expectations",
    batch_request=data_asset .build_batch_request()
)
validator.get_expectation_suite()
```
The output from this is
```
{
  "expectation_suite_name": "secret_expectations",
  "ge_cloud_id": null,
  "expectations": [
    {
      "expectation_type": "expect_column_values_to_match_regex",
      "kwargs": {
        "column": "secret",
        "regex": "\\d"
      },
      "meta": {}
    }
  ],
  "data_asset_type": null,
  "meta": {
    "great_expectations_version": "0.17.7"
  }
}
```
Notably, the `result_format` is missing from the kwargs.

**Comment by toivomattila** [comment]
Related issue: https://github.com/great-expectations/great_expectations/issues/8409

**Comment by austiezr** [comment]
Hey @toivomattila! Thanks for reaching out.

Result format is only persisted when set at the Checkpoint level -- GX doesn't currently support setting individual levels for result format on different expectations. When result format is set on an expectation, it is used to generate the results of running that expectation interactively with the validator, but isn't maintained beyond that, e.g. into an expectation suite or validation result.

I've captured this for our internal backlog to review this functionality, and I'll leave this Issue open for the next week. Please feel free to drop any further context or info about how you'd like to see this behavior change in the future!

**Comment by toivomattila** [comment]
Thank you for the quick response @austiezr !
I got everything to work nicely after changing the result format for Checkpoints

I'll add the case where this came up:

**TLDR**
As a developer creating an expectation suite, I wanted to use validation history to easily improve the expectation suite over time to decrease false positives.
This is not possible (afaik) if the dataset contains sensitive data that is validated but must not be in the reports.

**Longer version**
I have a dataset that contains some sensitive information that can't be in the reports (either Data Docs or Cloud)
I.e. a column with phone numbers
Most of the data in the dataset is fine and can be in the reports
I still want to validate the phone numbers and check for format, nulls, etc
If the checks fail, there can be only a "Failed" message in the reports, and none of the failed values to be included.
This works fine by setting the result format in the Checkpoint

Setting the result format for individual validations or even columns would be beneficial in order to get information about the other columns
I wrote an Expectation Suite for the other columns in the dataset but had access to only a sample/1 file so I made the Expectations (overtly) loose to reduce false negatives when new data was validated
This is suboptimal since it probably results in more false positives
My idea was to use the validation history in the reports to make the Expectations gradually tighter
For example, the expected row count for the files is currently unknown so I set the minimum required row count to 0 which is not realistic. I intended to increase the required row count after seeing from validations what the realistic values actually are.
Another example would be to set the allowed value range for a numerical column to be very wide and tighten it over time.
This doesn't work now since that information is not stored and available to me.

I can't just ask domain experts as no one in the company knows what would be good values for the expectations.
The files are deleted immediately after extracting the useful information so that we don't store any sensitive information

Anyway, this issue is fixed on my end and can be closed
